### DISCUSSION TITLE
Adding a Password

### DISCUSSION THREAD

**Discussion by gsandaru** [KiraLT/torrent-stream-server#246] [Q&A] [answered]
Hi,
Is there any way to add a master password to protect the web interface?

**Comment by KiraLT** [comment, marked as answer]
Yes, you can set a master password and you will be required to enter it when accessing User Interface or using API.

> Check [Security docs](https://github.com/KiraLT/torrent-stream-server/wiki/Security) for more information.

### Env variable

The easiest way to set API key is via [ENV variable](https://github.com/KiraLT/torrent-stream-server/wiki/configuration#env-variables):

```python
API_KEY="my password"
```
### Configuration file

You can also add `apiKey` to [configuration file](https://github.com/KiraLT/torrent-stream-server/wiki/configuration#file):

```json
{
    "security": {
        "apiKey": "my password",
    }
}
```

> To use the config file you will need to pass JSON config file to any run command with -c option (e.g. npm run start -c config.json).
